src/main/java/com/ldtteam/blockout/hooks/HookScreen.java

```diff
@@ -0,0 +1,88 @@
+package com.ldtteam.blockout.hooks;
+
+import com.ldtteam.blockout.BOScreen;
+import com.ldtteam.blockout.hooks.TriggerMechanism.Type;
+import com.ldtteam.blockout.views.ScrollingList;
+import com.mojang.blaze3d.matrix.MatrixStack;
+
+/**
+ * Screen wrapper.
+ */
+public class HookScreen extends BOScreen
+{
+    private final boolean captureScroll;
+    private ScrollingList scrollListener = null;
+
+    HookScreen(final HookWindow<?> window)
+    {
+        super(window);
+        captureScroll = window.windowHolder.hook.trigger.getType() == Type.RAY_TRACE;
+    }
+
+    public void render(final MatrixStack ms)
+    {
+        if (!isOpen)
+        {
+            return;
+        }
+
+        ms.translate(-width / 2, -height, 0.0d);
+        window.draw(ms, 0, 0);
+    }
+
+    public boolean mouseScrolled(final double scrollDiff)
+    {
+        if (captureScroll && scrollListener != null && scrollDiff != 0)
+        {
+            scrollListener.scrollInput(scrollDiff * 10, scrollListener.getX() + 1, scrollListener.getY() + 1);
+            return true; // TODO: would be nice to not stop event propagation when scrolling list is not scrollable
+        }
+        return false;
+    }
+
+    @Override
+    public void init()
+    {
+        // noop
+    }
+
+    @Override
+    public void tick()
```

Comment:
does this have to tick?

Reply:
the gui can have one scrolling list, so probably yes